pkg/minikube/cluster/cluster.go

```diff
@@ -78,6 +78,20 @@ func StartHost(api libmachine.API, config MachineConfig) (*host.Host, error) {
 	}
 
 	s, err := h.Driver.GetState()
+
+	// This is the name of the driver passed in
+	// through a flag, env, or config file (if any)
+	driverSpecified := config.VMDriver
+
+	// This is the name of the driver that is currently used with minikube
+	currentDriver := h.Driver.DriverName()
+
+	if driverSpecified != "" && driverSpecified != currentDriver {
```

Comment:
Yeah, we could try. Virtualbox definitely requires a restart, which I guess we could do here but it might be surprising.